crates/optimism/txpool/src/validator.rs

```diff
@@ -292,10 +297,10 @@ where
             )
             .await
         {
-            Ok(()) => Some(Ok(true)),
+            Ok(()) => Some(Ok(())),
```

Comment:
this match is redundant, so better use syntax `let Err(err) = client.validate_messages(..) else { return Ok(()) }` or `map_err`